### ISSUE TITLE
vctrs related error in R 4.0.0

### ISSUE THREAD

**Issue by nathancday** [r-box/boxr#160]
Haven't upgraded my local to 4.0 yet, but this error is being thrown by the Travis build:
![Screen Shot 2020-06-23 at 11 06 10 AM](https://user-images.githubusercontent.com/808225/85420860-be64cc00-b541-11ea-8227-e5abec57ef30.png)

It originates in `box_push()` and might be related to classing, see this [other issue](https://github.com/tidyverse/tidyr/issues/927)

**Comment by ijlyttle** [comment]
I'm getting the same error on my computer - I think it has to do with the internals of dplyr. I suspect we were getting away with something we shouldn't have.

It seems to be confined to one function:

<https://github.com/r-box/boxr/blob/7154b9f15a1802f173c0feca87787c8ab7056920/R/boxr__internal_verb_exit.R#L53-L62>

I can take a swing at it with a PR...

**Comment by ijlyttle** [comment]
@nathancday - would you consider a small PR, based on what you have already done, just for this?

**Comment by nathancday** [comment]
Totally it took my a few commits to figure out what was happening so it will be easiest to just grab the code. Maybe we can be specific `mutate_if(~ class(.) == "ocmode", ~as.character(.)) instead of my `mutate_all()` hammer

**Comment by ijlyttle** [comment]
From the PR perspective, it might be easiest (and I'm happy to do it tonight) to copy-paste your existing solution - in hopes of avoiding a merge-conflict when you pull these merged changes into the bigger PRs.

**Comment by ijlyttle** [comment]
Of course, I'm all for cleaning it up as soon as practicable.

**Comment by nathancday** [comment]
I just test this on this branch ☝️ and your `map_dfr` version works perfectly! I think that code is much cleaner and avoids my `mutate_all` hack. Shall we just use that?

I'll create a new "low-hanging fruit" branch. tomorrow after I respond your edits, so we can avoid merge conflicts later

**Comment by ijlyttle** [comment]
As I said in #163, I can get the tests to work, but I don't really *know* that it's doing the right thing - on the other hand I don't know who has ever used the return object... happy to go with it...

Once #163 is merged, you could run `usethis::pr_pull_upstream()` from your branch to get the changes - you'll have a merge conflict on this file, but it shouldn't be hard to sort out. Of course, you should do whatever makes sense to you.

**Comment by nathancday** [comment]
We might have to deal with another related error:
```
── 1. Error: box_fetch a dir (@test_07_fetch.R#21)  ────────────────────────────
Internal error in `vec_slice_impl()`: Unexpected `NULL`.
Backtrace:
 1. boxr::box_fetch("0", "test_dir", overwrite = FALSE, delete = FALSE)
 2. boxr:::fetchExit()
 3. boxr:::returnDwOp(...)
 4. purrr::map(...)
 5. boxr:::.f(.x[[i]], ...)
 6. purrr::map_dfr(op_detail$files, ~.x[item])
 7. dplyr::bind_rows(res, .id = .id)
 8. vctrs::vec_rbind(!!!dots, .names_to = .id)
```
Noticed this is my collab branch tests, after rebasing onto current master.

**Comment by nathancday** [comment]
I think it was closed via work in #155
